### PR TITLE
Support both V1 and V2

### PR DESCRIPTION
**Pull request by lillian542**
We rarely query the backends directly, and generally just pass them through to Qiskit. Current exceptions to this are checking the number of available qubits to verify against the number of wires, and getting the name of the backend for the Runtime Service execution method. 

This PR updates these two spots to work for both the new and the legacy Qiskit Backend API.

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-54390]

File: tests/test_base_device.py [hunk 1]
```diff
@@ -100,9 +130,46 @@ def options_for_testing():
...
+        
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=[0])
+            qml.CNOT(wires=[0, 1])
+            return qml.sample(qml.PauliZ(0))
+        
+        res = circuit(np.pi/2)
+        assert(isinstance(res, np.ndarray))
+        assert(np.shape(res) == (1024,))
```
Comment (obliviateandsurrender) [inline tests/test_base_device.py:172]:
```suggestion
        assert isinstance(res, np.ndarray)
        assert np.shape(res) == (1024,)
```

File: tests/test_qiskit_device.py [hunk 2]
```diff
@@ -13,6 +74,45 @@
...
+        
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=[0])
+            qml.CNOT(wires=[0, 1])
+            return qml.sample(qml.PauliZ(0))
+        
+        res = circuit(np.pi/2)
+        assert(isinstance(res, np.ndarray))
+        assert(np.shape(res) == (1024,))
```
Comment (obliviateandsurrender) [inline tests/test_qiskit_device.py:115]:
```suggestion
        assert isinstance(res, np.ndarray)
        assert np.shape(res) == (1024,)
```

**Review by obliviateandsurrender — COMMENTED** [review]
Thanks, looks great and should be good to go after a couple of minor suggested tweaks.

File: pennylane_qiskit/converter.py [hunk 3]
```diff
@@ -34,9 +35,6 @@
 
 import pennylane as qml
 import pennylane.ops as pennylane_ops
-from pennylane_qiskit.qiskit_device import QISKIT_OPERATION_MAP
-
-# pylint: disable=too-many-instance-attributes
```
[the remark this answers is not in the document]
Comment (austingmhuang) [reply pennylane_qiskit/converter.py:39]: Don't think we need this. I think I missed this when resolving merge conflicts (there were a lot).

**Review by austingmhuang — APPROVED** [review]
Fixed some of the minor issues @obliviateandsurrender caught. Thanks Utkarsh!

**Pull request merged**